builder/googlecompute/step_create_instance.go

```diff
@@ -24,13 +24,17 @@ func (c *Config) createInstanceMetadata(sourceImage *Image, sshPublicKey string)
 		instanceMetadata[k] = v
 	}
 
-	// Merge any existing ssh keys with our public key.
-	sshMetaKey := "sshKeys"
-	sshKeys := fmt.Sprintf("%s:%s", c.Comm.SSHUsername, sshPublicKey)
-	if confSshKeys, exists := instanceMetadata[sshMetaKey]; exists {
-		sshKeys = fmt.Sprintf("%s\n%s", sshKeys, confSshKeys)
+	// Merge any existing ssh keys with our public key, unless there is no
+	// supplied public key. This is possible if a private_key_file was
+	// specified.
+	if sshPublicKey != "" {
```

Comment:
There are cases where the public key won't be set, since the GCE project already has the public key in its metadata store. If we get rid of the check here, we'd need to move it down [here](https://github.com/mitchellh/packer/pull/4101/files#diff-7805d73f741f3122578fc04706a8a5b9R32) to avoid attaching malformed public key/username pairs to the instance.